### ISSUE TITLE
Version 1.x rewrite

### ISSUE THREAD

**Issue by milankinen** [tsers-js/core#13]
Hello Folks!

Based on the feedback I've received, I'm gonna do a **total re-write** of TSERS for version `1.x`. Don't be scared, that might sound bigger deal than it actually is. The goal of this re-write is to **simplify the API** surface and establish better and **more understandable terminology** without sacrificing the flexibility of the TSERS architecture.

The changes are following:
- Instead of **driver**, use **interpreter** 
  - The naming comes from [free monad pattern](http://underscore.io/blog/posts/2015/04/14/free-monads-are-simple.html) 
  - I don't want to restrict interpreter usage to only side-effects (which name "driver" implies), interpreters are only a way to interpret output signals from your application (e.g. in browser, virtual DOM is converted to actual DOM, in server it's converted to HTML string) 
- Rename **transducers** to (signal) **transformers** (the former sounds intimidating based on the feedback :smile:) 
- Simplify `main` signature from `main :: T => in$ => out$` to `main :: in$ => out$` 
  - The curried form was to complicated for beginners and didn't provide so much benefits in practice as I anticipated
  - `in$` is not a pure signal stream anymore but it contains both (possible) input$ signals and transformers by interpreter 
- Simplify interpreter signature from `interpreter :: Config => [signal$, T, executor]` to `interpreter :: Config => [signal$, executor]`
  - Again, `signal$` contains both input signals (if any) and signal transformers
- Make app initialization simpler: `const dispose = TSERS(main, { ...interpreters})`
- Rename common transforms:
  - `decompose` to `demux`
  - `compose` to `mux` 
  - `run` to `loop` 
- Remove `lift*` transforms 
- Add common transform: `mapListBy` based on [this](https://github.com/milankinen/stanga/#liftlistbyid) implementation
- Add [model interpreter](https://github.com/milankinen/stanga/#placing-the-state-to-model-driver-and-reading-it) and encourage its usage in examples/tutorials
- Re-write docs to use the new terminology and get rid of Cycle.js references

Each project is going to get `1.x` branch that will be merged into `master` once the re-write is complete

**Comment by milankinen** [comment]
Re-writing is in progress so if anyone has opinions or proposals, all on 'em are welcome. 

This is the best opportunity to propose breaking changes! :smile:

**Comment by milankinen** [comment]
Phew! Released!
